### DISCUSSION TITLE
PxParticleAndDiffuseBuffer - how to retrieve diffuse particle velocities?

### DISCUSSION THREAD

**Discussion by kresimirt** [NVIDIA-Omniverse/PhysX#220] [Q&A]
Hello,

Looking at the examples I can see that we can retrieve water particle positions and velocities.
However, for diffuse particles there seems to be only the call to get positions but not velocities?

PxParticleAndDiffuseBuffer* gParticleBuffer;

PxVec4* _gpupositions	= gParticleBuffer->getPositionInvMasses();
PxVec4* _gpuvelocities	= gParticleBuffer->getVelocities();

PxVec4* _gpudiffusePositions = gParticleBuffer->getDiffusePositionLifeTime();
//diffuse velocities?

Why is that since PhysX obviously keep those velocities internally.
Could we get such a method?

Thank you!

**Comment by sschirm** [comment]
Sorry for the very late response! This was actually added quite a while ago.
